nodeadm/internal/configprovider/userdata.go

```diff
@@ -85,6 +105,13 @@ func parseMultipart(userDataReader *multipart.Reader) (*internalapi.NodeConfig,
 				if err != nil {
 					return nil, err
 				}
+				if part.Header.Get(contentEncodingHeader) == "gzip" {
+					decompressedNodeConfigPart, err := decompressIfGZIP(nodeConfigPart)
+					if err != nil {
+						return nil, err
+					}
+					nodeConfigPart = decompressedNodeConfigPart
+				}
```

Comment:
Why do you need to make a check that it's `gzip` and then gracefully handle it not being `gzip`?

Reply:
wouldn't just calling `decompressIfGZIP` in all cases work better since its fallible in constant time and safe either way?

To not hit this branch you'd have to set the nodeConfig mediaType but forget to add the gzip encoding header, which im sure would only happen on accident. maybe handling it explicitly is more "proper", but not sure if its necessary

Reply:
MIME semantics dictate that if the section is GZIP'd, this header needs to be there. If a user (or library) is creating MIME documents that use GZIP but omit this header, that should fail because it's not conformant to the spec.

Generally, folks are only using GZIP compression via a library which should implement this properly -- folks aren't hand-writing and hand-compressing their MIME docs.

Reply:
Turns out cloudinit doesn't care what the spec says, and just tries to base64-decode and gzip-decompress pretty much anything it handles at any scope. It's more important for us to match the cloudinit behavior than to follow the spec, for better or worse. Updated in latest rev